streams/src/main/java/org/apache/kafka/streams/processor/internals/PartitionGroup.java

```diff
@@ -93,8 +93,8 @@ RecordQueue queue() {
                    final Map<TopicPartition, RecordQueue> partitionQueues,
                    final Function<TopicPartition, OptionalLong> lagProvider,
                    final Sensor recordLatenessSensor,
+                   final Sensor totalInputBufferBytesSensor,
                    final Sensor enforcedProcessingSensor,
-                   final Sensor totalBytesSensor,
```

Comment:
Thinking about this fix: would it be possible to avoid such errors in the future? Can we make `Sensor` more type save? -- For this issue, do we have missing tests (or it this too hard to test)?

Reply:
There are a couple of unit tests: https://github.com/apache/kafka/blob/0924fd3f9f75c446310ed1e97b44bbc3f33c6c31/streams/src/test/java/org/apache/kafka/streams/processor/internals/PartitionGroupTest.java#L780 which try to check. I think whats missing- and it's a miss on my part which I now realise- is an e2e IT. Something like push some data, check paused partitions and later on test their resumption.

Reply:
Hello @vamossagar12 could you file a JIRA/PR for the test coverage?

Reply:
@guozhangwang , done.. Please find the JIRA here: https://issues.apache.org/jira/browse/KAFKA-14040